pkg/ccl/backupccl/restore_job.go

```diff
@@ -221,6 +223,65 @@ func makeBackupLocalityMap(
 	return backupLocalityMap, nil
 }
 
+func filterCompletedImportSpans(
+	completedSpansSlice []jobspb.RestoreProgress_RestoreProgressFrontierEntry,
+	importSpans []execinfrapb.RestoreSpanEntry,
+) (*spanUtils.Frontier, []execinfrapb.RestoreSpanEntry, error) {
+	var frontierCreationSlice []roachpb.Span
+	var modifiedImportSpans []execinfrapb.RestoreSpanEntry
+	for _, frontierSpanEntry := range completedSpansSlice {
+		frontierCreationSlice = append(frontierCreationSlice, frontierSpanEntry.Entry)
+	}
+	checkpointingSpanFrontier, err := spanUtils.MakeFrontier(frontierCreationSlice...)
+	if err != nil {
+		return checkpointingSpanFrontier, modifiedImportSpans, err
+	}
+	for _, completedSpanEntry := range completedSpansSlice {
+		_, err = checkpointingSpanFrontier.Forward(completedSpanEntry.Entry, completedSpanEntry.Timestamp)
+		if err != nil {
+			return checkpointingSpanFrontier, modifiedImportSpans, err
+		}
+	}
+
+	for _, importSpan := range importSpans {
+		skip := false
+		checkpointingSpanFrontier.SpanEntries(importSpan.Span, func(s roachpb.Span,
+			ts hlc.Timestamp) (done spanUtils.OpResult) {
+			if ts.Equal(hlc.Timestamp{WallTime: 1}) {
+				skip = true
+				return spanUtils.ContinueMatch
+			}
+			skip = false
+			return spanUtils.StopMatch
+		})
+		if skip {
```

Comment:
nit: reduce the amount of branching by:
```
if !skip{
  modifiedImportSpans = append(modifiedImportSpans,importSpan)
}
```

Reply:
I believe this isn't possible, checking the skip condition is to skip over the the given span so that it is not included in toDoSpans

Reply:
i'm not following, but this is a small nit we can chat about during our check in.